apps/console/src/features/users/components/wizard/bulk-import-user-wizard.tsx

```diff
@@ -152,17 +178,81 @@ export const BulkImportUserWizard: FunctionComponent<BulkImportUserInterface> =
     const [ bulkResponseSummary, setBulkResponseSummary ] = useState<BulkResponseSummary>(initialBulkResponseSummary);
     const [ readWriteUserStoresList, setReadWriteUserStoresList ] = useState<DropdownItemProps[]>([]);
     const [ selectedUserStore, setSelectedUserStore ] = useState<string>("");
-    
-    useEffect(() => {
-        setSelectedUserStore(userstore);
-        getUserStoreList();
-    }, [ userstore ]);
     const [ configureMode, setConfigureMode ] = useState<string>(undefined);
     const [ emailData, setEmailData ] = useState<string[]>();
     const [ isEmailDataError, setIsEmailDataError ] = useState<boolean>(false);
     const [ emailDataError, setEmailDataError ] = useState<string>("");
+    const [ roleUserAssociations, setRoleUserAssociations ] = useState<Record<string, RoleUserAssociation>>({});
 
     const optionsArray: string[] = [];
+    
+    const {
+        data: rolesList,
+        isLoading: isRolesListLoading,
+        error: rolesListError
+    } = useRolesList(
+        undefined, undefined, ORG_ROLE_FILTER
+    );
+    
+    /**
+     * Handle if any error occurs while fetching the roles list.
+     */
+    useEffect(() => {
+        if (rolesListError) {
+            dispatch(
+                addAlert({
+                    description: t("console:manage.features.roles.notifications.fetchRoles.genericError.description"),
+                    level: AlertLevels.ERROR,
+                    message: t("console:manage.features.roles.notifications.fetchRoles.genericError.message")
+                })
+            );
+        }
+    }, [ rolesListError ]);
+    
+    /**
+     * Fetch the user roles list.
+     */
+    useEffect(() => {
+        const newRoles: Record<string, RoleUserAssociation> = {};
+    
+        rolesList?.Resources?.map((role: RolesInterface) => {
+            newRoles[role.displayName.toLowerCase()] = {
+                displayName: role.displayName,
+                id: role.id,
+                users: []
+            };
+        });
+    
+        setRoleUserAssociations(newRoles);
+    }, [ rolesList ]);
+    
+    /**
+     * Set the user store list.
+     */
+    useEffect(() => {
+        setSelectedUserStore(userstore);
+        getUserStoreList();
+    }, [ userstore ]);
+
+    const getGroupMemberAssociation = async (): Promise<Record<string, GroupMemberAssociation>> => {
+        try {
+            const response: any = await getGroupList(null);
+            const newGroups: Record<string, GroupMemberAssociation> = {};
+
+            response.data.Resources.map((group: GroupsInterface) => {
+                newGroups[group.displayName.toLowerCase()] = {
+                    displayName: group.displayName,
+                    id: group.id,
+                    members: []
+                };
+            });
```

Comment:
Can be improved with `reduce` method instead of `map`. More appropriate to transform an array into a single value. Try following,

```
const newGroups = response.data.Resources.reduce((acc: Record<string, GroupMemberAssociation>, group: GroupsInterface) => {
    acc[group.displayName.toLowerCase()] = {
        displayName: group.displayName,
        id: group.id,
        members: []
    };
    return acc;
}, {});
```

Reply:
Improved